gravitee-apim-console-webui/src/management/application/details/subscribe/application-subscribe.controller.ts

```diff
@@ -134,6 +150,25 @@ class ApplicationSubscribeController {
         )),
     );
   }
+
+  selectKeyMode() {
+    const dialog = {
+      controller: 'ApplicationKeyDialogController',
+      controllerAs: '$ctrl',
+      template: require('./application-key.dialog.html'),
+      clickOutsideToClose: true,
+    };
+
+    return this.$mdDialog.show(dialog);
+  }
+
+  get shouldPromptForKeyMode(): boolean {
+    return this.allowsSharedApiKeys && this.subscribedAPIs.length >= 1 && this.application.api_key_mode === ApiKeyMode.UNSPECIFIED;
```

Comment:
This checks `subscribedAPIs.length`, which, I think, can contain subscriptions other than API keys (for example Oauth2).
Should we add something to consider only API key subscriptions ?